# AFFLUENT FILMS AT BOX OFFICE : CASE STUDY MICROSOFT


## 1. Business Understanding

Microsoft is  a technology corporation best known for software products such as Windows operating system. The company would like to create a movie studio due to the trending original video creation. In order to do this,the company needs to know the type of films that are currently doing the best at box office. 

This project will extract the relevant data and provide insight on the trending films by providing meaningfull visualizations. It will also recommend on the type of films the company should venture into. The project will also show the budget associated with each film and recommend on whether or not the company should venture into that film. The project will also recommend on whether or not the company should venture into filming at all based on the net returns of the films.

## 2.Data Understanding

The data used in this project was extracted from [IMDb](https://www.imdb.com/) ,[Box Office Mojo](https://www.boxofficemojo.com/) ,[The Numbers](https://www.the-numbers.com/) ,[TheMovieDB](https://www.themoviedb.org/).

The dataset from Box Office Mojo contains the gross income of each movie,both domestic and foreign and the year it was released. This helps in determining the income generated from each movie. This dataset contains 3387 rows and 5 columns.

The IMDb website contained multiple datasets. The following datasets were extracted:
- crew dataset : This dataset contains the movie ids and their corresponding directors and writters. It can be joined with other datasets to yield important information such as which directors direct movies with the highest  ratings ?

- title and ratings dataset : This dataset contains the code names of movies their rating and number of people who voted. It can be of use to determine whether a certain film was popular(loved) or hated.

- title and basics dataset : This dataset contains the names and code names of movies and their genres. 

The dataset extracted from The Numbers website contains the names of movies and their associated budgets as well as the domestic and worldwide gross. It can help determine losses associated with each movie as well as the profits.

The dataset extracted from TheMovieDB contains several columns that can be of great use to this project. It will help us answer questions such as which genres had the highest rating ?

The project will use these datasets to come up with the insights and relevant recommendations.

##  3.Data preparation

### Importing the relevant libraries

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

%matplotlib inline
sns.set()

### Loading and exploring the various data sets

In [2]:
# Loading the box office mojo datasets
bom_data_set = pd.read_csv('bom.movie_gross.csv.gz')
bom_data_set.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [3]:
# loading the imdb title and basics dataset
title_basics = pd.read_csv('imdb.title.basics.csv.gz')
title_basics.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [4]:
# Loading the imdb title and ratings dataset
ratings_df= pd.read_csv('imdb.title.ratings.csv.gz')
ratings_df.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [5]:
# Loading the crew dataset from imdb
crew_df= pd.read_csv('imdb.title.crew.csv.gz')
crew_df.head()

tconst                      directors              writers
0  tt0285252                      nm0899854            nm0899854
1  tt0438973                            NaN  nm0175726,nm1802864
2  tt0462036                      nm1940585            nm1940585
3  tt0835418                      nm0151540  nm0310087,nm0841532
4  tt0878654  nm0089502,nm2291498,nm2292011            nm0284943

In [6]:
# Loading the movies_df from TheMovieDB
tmdb_df =pd.read_csv('tmdb.movies.csv.gz')
tmdb_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [7]:
#Loading the dataset from The Numbers
numbers_df = pd.read_csv('tn.movie_budgets.csv.gz')
numbers_df

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

###  Data wrangling

### Checking for missing values and duplicates in each dataset

In [8]:
# Determining whether bom dataset has missing values 
bom_data_set.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

It can be seen that the column 'studio' 5 missing values,'foreign_gross' column has 1350 missing values and 'domestic_gross' has 28 missing values. 

In [9]:
# CHecking the shape of the data set to determine the number of rows(length of the data set)
bom_data_set.shape

(3387, 5)

In [10]:
# checking the percentage of missing values in the 'domestic_gross' column
miss = bom_data_set['foreign_gross'].isna().sum() # calculates the number of missing values in the column
total_rows = len(bom_data_set)  # length ot the series
miss_p = (miss/total_rows)* 100 # percentage of missing values
miss_p

39.85828166519043

The number of missing values in the bom dataset is too high to be dropped. However the numbers_df containing the dataset from The Numbers contains almost similar information.

In [11]:
# Exploring the numbers_df for missing values
numbers_df.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

The dataset contains no missing values

In [12]:
# Exploring the data types in the numbers_df 
numbers_df.dtypes

id                    int64
release_date         object
movie                object
production_budget    object
domestic_gross       object
worldwide_gross      object
dtype: object

The data types are as anticipated.

In [13]:
# Checking for duplicates in the numbers_df dataset
# setting the subsets to release_date and movies.. a movie might have the same name but different release dates
numbers_df.duplicated(subset = ['release_date','movie']).sum() 

0

The numbers_df dataset has no duplicates.

In [14]:
# Checking whether the ratings_df has missing values
ratings_df.isna().sum()

tconst           0
averagerating    0
numvotes         0
dtype: int64

The dataset has no missing values

In [15]:
# Checking the data types in the ratings_df
ratings_df.dtypes

tconst            object
averagerating    float64
numvotes           int64
dtype: object

The data types are as expected..numeric

In [16]:
# checking for duplicates
# Subset is set to tconst i.e title ids
ratings_df.duplicated(subset = 'tconst').sum()

0

The ratings_df has no duplicates

In [17]:
# Checking for missing values the title_basics dataset
title_basics.isna().sum()

tconst                 0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

In [18]:
# checking the shape of the title_basics dataset
title_basics.shape

(146144, 6)

The 'runtime_minutes' column contains the highest number of missing values. Dropping this column might limit the scope of our research. It would be wise to replace the missing values in the dataset with a measure of central tendecy. We will use the median since it is least affected by outliers

In [19]:
# Replacing null run_rime rows in the title_basics dataframe with the median
title_basics['runtime_minutes'].fillna(title_basics['runtime_minutes'].mean(),inplace = True)
title_basics.isna().sum()

tconst                0
primary_title         0
original_title       21
start_year            0
runtime_minutes       0
genres             5408
dtype: int64

The remaining missing rows with missing values in the title_basics dataframe can be dropped..They do not form a large part of the dataset.

In [20]:
title_basics.dropna(inplace = True)
title_basics.isna().sum() # Checking for missing values to confirm

tconst             0
primary_title      0
original_title     0
start_year         0
runtime_minutes    0
genres             0
dtype: int64

In [21]:
title_basics.shape # Checking the shape of the dataframe

(140734, 6)

In [22]:
# checking for duplicates in the title_basics dataset
# Setting the subset to tconst...title ids should be unique
title_basics.duplicated(subset = 'tconst').sum()

0

The title_basics dataset contains no duplicates

In [23]:
# Checking the data types of the title_basics dataset
title_basics.dtypes

tconst              object
primary_title       object
original_title      object
start_year           int64
runtime_minutes    float64
genres              object
dtype: object

The data types are as expected

In [24]:
# Checking for missing values in the crew_df
crew_df.isna().sum()

tconst           0
directors     5727
writers      35883
dtype: int64

In [25]:
crew_df.shape

(146144, 3)

The 'writers' column in the crew_df contains a large number of missing values. The project can still work with the directors to provide any insight concerning the personnel. The column can therefore be dropped.The 'directors' column contains a small number of missing values compared to the large number of rows (146144). the missing values can be dropped since it is a small portion of the dataset.

In [26]:
# Dropping the 'writers' column from crew_df
crew_df.drop('writers',axis = 1,inplace = True)
crew_df.shape # checking the number of columns

(146144, 2)

In [27]:
# Removing the missing values from the 'directors' column
crew_df.dropna(inplace = True)
crew_df.isna().sum() #checking for missing values

tconst       0
directors    0
dtype: int64

The dataset now has no missing values

In [28]:
# Checking for duplicates in the crew_df
# Subset is set to the movie ids i.e tconst
crew_df.duplicated(subset = 'tconst').sum()

0

The crew dataset contains no duplicates

In [29]:
# Checking the data types in the crew_df
crew_df.dtypes

tconst       object
directors    object
dtype: object

The data types of the two remaining columns are as expected

In [30]:
# Checking for missing values in the tmdb_df 
tmdb_df.isna().sum()

Unnamed: 0           0
genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64

The dataset has no missing values

In [31]:
# Checking for duplicates in the tmdb_df
# Setting subset to id since they should be unique
tmdb_df.duplicated(subset ='id').sum()

1020

In [32]:
# Checking the number of unique values...items with duplicates will have a frequency greater than one
tmdb_df['id'].value_counts()

380718    3
292086    3
402448    3
192137    3
514791    3
         ..
174376    1
221791    1
160165    1
250251    1
309885    1
Name: id, Length: 25497, dtype: int64

In [33]:
# Filtering to obtain the datasets that have an id of 380718
tmdb_df[tmdb_df['id'] ==380718 ]

Unnamed: 0            genre_ids      id original_language  \
17256       17256  [12, 35, 18, 10770]  380718                en   
20494       20494  [12, 35, 18, 10770]  380718                en   
23724       23724  [12, 35, 18, 10770]  380718                en   

            original_title  popularity release_date                title  \
17256  The Ultimate Legacy         0.6   2015-12-04  The Ultimate Legacy   
20494  The Ultimate Legacy         0.6   2015-12-04  The Ultimate Legacy   
23724  The Ultimate Legacy         0.6   2015-12-04  The Ultimate Legacy   

       vote_average  vote_count  
17256           3.8           6  
20494           3.8           6  
23724           3.8           6

From the data_set above it can be stated that the movies with the same id are the same since the release_dates are also the same.

In [34]:
# Droping the duplicates based on the id
tmdb_df.drop_duplicates(subset = 'id',inplace = True)
tmdb_df.duplicated(subset ='id').sum() # Checking for duplicates

0

In [35]:
# Dropping the 'Unnamed: 0' columns from the tmdb_df
tmdb_df.drop('Unnamed: 0',axis = 1,inplace =  True)

In [36]:
# Checking the data types in tmdb_df
tmdb_df.dtypes

genre_ids             object
id                     int64
original_language     object
original_title        object
popularity           float64
release_date          object
title                 object
vote_average         float64
vote_count             int64
dtype: object

The data types match the expectations for each column